### ISSUE TITLE
I can't find `title` attribute in version 4.0

### ISSUE THREAD

**Issue by AionDev** [mdgriffith/style-elements#84]
`title` attribute is nowhere to be found in version 4.0. 

Was this intentional or an oversight? 

For now, I can use  `attribute "title" "shows on hover.."` - maybe not as nice. 

I argue `title` should be included because is fairly common, and I can misspell `attribute "tile"  "shows on hover.."` and then no errors will show up. 
Thank you :)

**Comment by mdgriffith** [comment]
I removed a lot of stuff that was inherited from the `html` library because I want to make sure that things that are included in the style elements lib are designed deliberately instead of just inheriting from html.

`title` could be a good candidate to be added back in though, can you give a brief explanation on the situations you use it in?

**Comment by jxxcarlson** [comment]
Yes, here is a typical example:

In module Views.Basic:

faIcon title_ icon attrs =
    el NavBarActive
        ([ height (px 25), width (px 20), title title_,  alignBottom, verticalCenter ] ++ attrs)
        (iconWhite20 icon)

I use this function extensively to pace icons in my app, e.g., 

randomDocumentIcon : Model -> Element Styles variation Msg
randomDocumentIcon model =
    Basic.faIcon "Get random documents" FontAwesome.random [ onClick RandomDocuments ]

This displays the FontAwesome.random icon as an active element that when clicked displays
a list of random documents.  When the user hovers over the icon, the the text “Get random documents”
is displayed.  Because of the way the style NavBarActive is defined, the icon “blinks” when clicked.

BTW, the above shows why Elm + Style-Elements is all one needs to make “Components.”

— Jim

**Comment by AionDev** [comment]
yep! same here, i'm using title to explain icons: 
![i use title to explain what icons do](https://user-images.githubusercontent.com/20288140/30564976-633cc19e-9ccf-11e7-9e9b-7530ff860ac5.png)

It's annoying to hover over something and not understand what is suppose to do. `That icon is for showing the computation pipeline` in my debugger.

**Comment by cmditch** [comment]
Agreed. `title` would be nice to have. 

@mdgriffith thanks btw, loving the library so far. So much more pleasant than dealing with CSS.

**Comment by mdgriffith** [comment]
OK, so what I'm hearing is that what we really want is someway to conveniently create a tooltip :)

If I added a way to specify tooltips to the lib (`el`s that are shown when you hover on the tagged box), would that handle the basic usecase that people are using `title` for?

**Comment by jxxcarlson** [comment]
That is what I was using it for.

**Comment by AionDev** [comment]
![tooltip in github](https://user-images.githubusercontent.com/20288140/30776969-18f2f9e4-a0b9-11e7-93df-dc059c974c46.png)

If we would easily create something like this would be so much better. `Title` is pretty bad design compared with this 😄

**Comment by kuon** [comment]
Yeah I agree that proper tooltip would be nice, but it should also handle title for accessibility.

**Comment by jesse-c** [comment]
Titles are also used for accessibility. [1] If a custom tooltip implementation was added, those users may still be missing out.

[1] https://www.w3.org/WAI/tutorials/forms/labels/
